shell/server/proxy.js

```diff
@@ -1620,8 +1614,96 @@ Proxy.prototype.handleRequest = function (request, data, response, retryCount) {
       }, context);
     } else if (request.method === "DELETE") {
       return session.delete(path, context);
+    } else if (request.method === "PROPFIND") {
+      return session.propfind(path, {
+        mimeType: request.headers["content-type"] || "application/octet-stream",
+        content: data,
+        encoding: request.headers["content-encoding"]
+      }, context);
+    } else if (request.method === "PROPPATCH") {
+      return session.proppatch(path, {
+        mimeType: request.headers["content-type"] || "application/octet-stream",
```

Comment:
Good point. My recollection of the webdav spec is that the client SHOULD pass a content-type header for all requests with bodies (most likely application/xml) but I don't think we need to default to application/octet-stream. We should probably let the grain handle unknown content-types instead. I've extracted all of it into its own method so it's a little easier to read.